source/components/Autocomplete.js

```diff
@@ -192,24 +196,44 @@ class AutocompleteBase extends Component<AutocompleteProps, State> {
       !maxSelections ||
       (canMoreOptionsBeSelected && areFilteredOptionsAvailable)
     ) {
-      if (!selectedOption) return;
-      const option = selectedOption.trim();
-      const optionCanBeSelected =
-        (selectedOptions.indexOf(option) < 0 && !multipleSameSelections) ||
-        multipleSameSelections;
-
-      if (option && optionCanBeSelected && isOpen) {
-        const newSelectedOptions = _.concat(selectedOptions, option);
-        this.selectionChanged(newSelectedOptions, event);
-        this.setState({ selectedOptions: newSelectedOptions, isOpen: false });
+      if (!selectedOption || !selectedOption.length) return;
+      const option = _.isString(selectedOption) ? selectedOption.trim() : selectedOption;
+      let newSelectedOptions = [];
+      let skipValuesSelection = false;
+      if (option && Array.isArray(option)) {
+        newSelectedOptions = selectedOptions;
+        option.forEach(item => {
+          const optionCanBeSelected =
+            (selectedOptions.indexOf(item) < 0 &&
+              newSelectedOptions.indexOf(item) < 0 &&
+              filteredOptions.indexOf(item) &&
+              !multipleSameSelections) ||
+            multipleSameSelections;
+          if (filteredOptions.indexOf(item) < 0) {
+            skipValuesSelection = true;
+            return;
+          }
+          if (item && optionCanBeSelected && isOpen && !skipValuesSelection) {
+            newSelectedOptions = _.concat(newSelectedOptions, item);
+          }
+        });
+      } else {
+        const optionCanBeSelected =
+          (selectedOptions.indexOf(option) < 0 && !multipleSameSelections) ||
+          multipleSameSelections;
```

Comment:
can be simplified to `multipleSameSelections || !selectedOptions.includes(option)`

Reply:
Fixed